tests/hypothesis_strategies/constraints.py

```diff
@@ -0,0 +1,149 @@
+"""Hypothesis strategies for constraints."""
+
+from functools import partial
+from typing import Any, List, Optional, Type, Union
+
+import hypothesis.strategies as st
+
+from baybe.constraints.conditions import (
+    SubSelectionCondition,
+    ThresholdCondition,
+    _valid_logic_combiners,
+)
+from baybe.constraints.continuous import (
+    ContinuousLinearEqualityConstraint,
+    ContinuousLinearInequalityConstraint,
+)
+from baybe.constraints.discrete import (
+    DiscreteExcludeConstraint,
+    DiscreteLinkedParametersConstraint,
+    DiscreteNoLabelDuplicatesConstraint,
+    DiscreteProductConstraint,
+    DiscreteSumConstraint,
+)
+from baybe.parameters.base import DiscreteParameter
+from baybe.parameters.numerical import NumericalDiscreteParameter
+
+
+def sub_selection_conditions(superset: Optional[List[Any]] = None):
+    """Generate :class:`baybe.constraints.conditions.SubSelectionCondition`."""
+    if superset is None:
+        element_strategy = st.text()
+    else:
+        element_strategy = st.sampled_from(superset)
+    return st.builds(SubSelectionCondition, st.lists(element_strategy, unique=True))
+
+
+def threshold_conditions():
+    """Generate :class:`baybe.constraints.conditions.ThresholdCondition`."""
+    return st.builds(
+        ThresholdCondition, threshold=st.floats(allow_infinity=False, allow_nan=False)
```

Comment:
this would create threshold conditions that dont make sense in extreme cases, like when the `<` threshold is lower than the smallest numerical element, is there an easy way to include that or do we even intend this thing to be tested too (in practice nothing would happen as the filtering has no effect)? It could also have the effect that the entire searchspace is deleted etc (exclude constraint with > smallest element threshold)

Reply:
If you ask me, I think this is perfectly fine FOR NOW. I don't consider these strategies final anyway, we can always adjust or rather extend when needed. As you also already mentioned in a different thread: in the end, they need to be adjusted such that we can cover a broad range of meaningful search spaces as well as edge cases. So I think the moment to discuss this is not actually in this PR, but rather in the hypothesis PR for the searchspaces. We can then easily parametrize the strategy and add sensible defaults 👍🏼 

Will resolve but feel free to reopen if necessary